### PR TITLE
Hotfixes and Packages commands added to wazuhDBQueryBuilder

### LINKED ISSUE #20201: WazuhDB queries to fetch hotfixes and packages with cached command and due
## Description
Add necessary Wazuh-DB wrapper to manage requests to obtain the agent's hotfixes and packages.

Currently, the Wazuh-DB query builder only uses the cached command for the OS info. Using the full `sql` command for the hotfixes might:
- Be slower because it doesn't use the cache
- It can fail if we have too many hotfixes.

The Wazuh-DB daemon already has a helper that overcomes these limitations and the old vulnerability detector uses.
We should implement the same mechanism for the wrapper.

The "due" responses must be handled on the class SocketDBWrapper.
For this kind of response, the resulting JSON response will be an object array with all the items received from the Wazuh-DB daemon.

For example, for the following response:
```
matias@matias-X580VD:~/Downloads$ python3 wdb-query.py 2 hotfix get
due {"hotfix":"KB2468871"}
due {"hotfix":"KB2478063"}
due {"hotfix":"KB2533523"}
due {"hotfix":"KB2544514"}
due {"hotfix":"KB2600211"}
due {"hotfix":"KB2600217"}
due {"hotfix":"KB5014032"}
due {"hotfix":"KB5015684"}
due {"hotfix":"KB5016705"}
due {"hotfix":"KB5026037"}
due {"hotfix":"KB5026361"}
due {"hotfix":"KB5026435"}
due {"hotfix":"KB5026513"}
ok {"status":"SUCCESS"}
```
The response JSON object for SocketDBWrapper should be:
```
[{"hotfix":"KB2468871"},{"hotfix":"KB2478063"},{"hotfix":"KB2533523"},{"hotfix":"KB2544514"},{"hotfix":"KB2600211"},{"hotfix":"KB2600217"},{"hotfix":"KB5014032"},{"hotfix":"KB5015684"},{"hotfix":"KB5016705"},{"hotfix":"KB5026037"},{"hotfix":"KB5026361"},{"hotfix":"KB5026435"},{"hotfix":"KB5026513"}]
```

## DoD

- [x] The vulnerability scanner module is able to request the required information from Wazuh-DB
- [x] UTs added
- [x] CI Pass

### PR DESCRIPTION
**Pull request by matias-braida**
Wazuh db responses with due support added to socketDBWrapper

|Related issue|
|---|
|Closes #20201|

## Description

This PR adds the command "hotfix" and "packages" to wazuhDBQueryBuilder.
Also, Wazuh-DB responses with due support were added to socketDBWrapper class.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] run Wazuh-DB request of the new commands supported

### CODE REVIEW

File: src/shared_modules/utils/socketDBWrapper.hpp [hunk 1]
```diff
@@ -43,8 +45,49 @@ class SocketDBWrapper final
...
+                    m_conditionVariable.notify_one();
+                }
+                else if (0 == responsePacket.compare(0, 3, DB_WRAPPER_UNKNOWN))
+                {
+                    m_exceptionStr = "DB query unknown response: " + responsePacket.substr(4);
+                    m_conditionVariable.notify_one();
+                }
+                else if (0 == responsePacket.compare(0, 3, DB_WRAPPER_DUE))
+                {
+                    m_responsePartial.push_back(nlohmann::json::parse(responsePacket.substr(4)));
```
Comment (pereyra-m) [inline src/shared_modules/utils/socketDBWrapper.hpp:72]: The exceptions that `nlohmann::json::parse()` could throw aren't being handled.

File: src/shared_modules/utils/socketDBWrapper.hpp [hunk 2]
```diff
@@ -43,8 +45,49 @@ class SocketDBWrapper final
...
+                }
+                else if (0 == responsePacket.compare(0, 2, DB_WRAPPER_OK))
+                {
+                    if(!m_responsePartial.empty())
+                    {
+                        m_response = m_responsePartial;
+                    }
+                    else
+                    {
+                        m_response = nlohmann::json::parse(responsePacket.substr(3));
```
Comment (pereyra-m) [inline src/shared_modules/utils/socketDBWrapper.hpp:82]: The exceptions that `nlohmann::json::parse()` could throw aren't being handled.

File: src/shared_modules/utils/socketDBWrapper.hpp [hunk 3]
```diff
@@ -43,8 +45,49 @@ class SocketDBWrapper final
             [&](const char* body, uint32_t bodySize, const char* header, uint32_t headerSize)
             {
                 std::unique_lock<std::mutex> lock {m_mutex};
-                m_response = std::string(body, bodySize);
-                m_conditionVariable.notify_one();
+                std::string responsePacket(body, bodySize);
+
+                if (responsePacket.empty())
```
Comment (pereyra-m) [inline src/shared_modules/utils/socketDBWrapper.hpp:50]: Coding style: this is only a suggestion.
We could avoid writing `m_conditionVariable.notify_one()` every time is we simply change the order

<details><summary>Details</summary>
<p>

```c++
                if (0 == responsePacket.compare(0, 3, DB_WRAPPER_DUE)) {
                    m_responsePartial.push_back(nlohmann::json::parse(responsePacket.substr(4)));
                } else {
                    if (responsePacket.empty())
                    {
                        m_exceptionStr = "Empty DB response";
                    }
                    else if (0 == responsePacket.compare(0, 3, DB_WRAPPER_ERROR))
                    {
                        m_exceptionStr = "DB query error: " + responsePacket.substr(4);
                    }
                    else if (0 == responsePacket.compare(0, 3, DB_WRAPPER_IGNORE))
                    {
                        m_exceptionStr = "DB query ignored: " + responsePacket.substr(4);
                    }
                    else if (0 == responsePacket.compare(0, 3, DB_WRAPPER_UNKNOWN))
                    {
                        m_exceptionStr = "DB query unknown response: " + responsePacket.substr(4);
                    }
                    else if (0 == responsePacket.compare(0, 2, DB_WRAPPER_OK))
                    {
                        if(!m_responsePartial.empty())
                        {
                            m_response = m_responsePartial;
                        }
                        else
                        {
                            m_response = nlohmann::json::parse(responsePacket.substr(3));
                        }
                    }
                    else
                    {
                        m_exceptionStr = "DB query invalid response: " + responsePacket;
                    }
                    m_conditionVariable.notify_one();
                }
```



</p>
</details>

**Review by pereyra-m — CHANGES_REQUESTED** [review]
Gj!

I've made a small test with the packages just to verify the behavior.

This is the `config.json` used

```
{
  "queries": [
    [
      {
        "method": "agentGetPackagesCommand",
        "arg": "0"
      }
    ],
    [
      {
        "method": "agentGetHotfixesCommand",
        "arg": "0"
      }
    ]
  ]
}
```


It seems to work, but just in the end, the two outputs are being mixed:

[out.txt](https://github.com/wazuh/wazuh/files/13385291/out.txt)

```
...
    {
        "architecture": "amd64",
        "item_id": "ffed5ae9522c143f9a01f3f5b32b6567484ad9e7",
        "name": "libepoxy0",
        "source": "libepoxy",
        "vendor": "Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>",
        "version": "1.5.10-1"
    }
]
Response to "agent 0 hotfix get ":
[
    {
        "architecture": "amd64",
        "item_id": "0071f0293f6d11217e064490e235c7393b6aed13",
        "name": "libatspi2.0-0",
        "source": "at-spi2-core",
        "vendor": "Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>",
        "version": "2.44.0-3"
    },
    {
        "architecture": " ",
        "item_id": "0090a25615e066aaaae96bf36a13bc44afc243bb",
        "name": "certifi",
        "source": " ",
        "vendor": " ",
        "version": "2022.12.7"
    },
...
```

But the packages query is out of scope, so I wouldn't worry about this now.

File: src/shared_modules/utils/socketDBWrapper.hpp [hunk 4]
```diff
@@ -30,7 +30,9 @@ class SocketDBWrapper final
 {
 private:
     std::shared_ptr<SocketClient<Socket<OSPrimitives, SizeHeaderProtocol>, EpollWrapper>> m_dbSocket;
-    std::string m_response;
+    nlohmann::json m_response;
+    nlohmann::json m_responsePartial;
+    std::string m_exceptionStr;
```
[the remark this answers is not in the document]
Comment (matias-braida) [reply src/shared_modules/utils/socketDBWrapper.hpp:35]: fixed

File: src/shared_modules/utils/socketDBWrapper.hpp (hunk 1 above)
Comment (matias-braida) [reply src/shared_modules/utils/socketDBWrapper.hpp:72]: fixed

File: src/shared_modules/utils/socketDBWrapper.hpp (hunk 2 above)
Comment (matias-braida) [reply src/shared_modules/utils/socketDBWrapper.hpp:82]: fixed

File: src/wazuh_modules/vulnerability_scanner/testtool/wazuhDBQuery/main.cpp
Comment (pereyra-m) [inline src/wazuh_modules/vulnerability_scanner/testtool/wazuhDBQuery/main.cpp:1]: Just a detail:
I've run the testtool, and I've found that all the responses are arrays (even with no elements), except by the `hotfix` request when there are no hotfixes

```
Response to "agent 0 hotfix get ":
{
    "status": "SUCCESS"
}
```

We should standardize the output so the user always receives an array.

**Review by pereyra-m — COMMENTED** [review]
Gj!

File: src/wazuh_modules/vulnerability_scanner/testtool/wazuhDBQuery/main.cpp
Comment (matias-braida) [reply src/wazuh_modules/vulnerability_scanner/testtool/wazuhDBQuery/main.cpp:1]: fixed

**Review by pereyra-m — APPROVED** [review]
LGTM!

**Pull request merged**